pkg/apis/controlplane/v1beta2/types.go

```diff
@@ -389,6 +389,10 @@ type NetworkPolicyNodeStatus struct {
 	NodeName string `json:"nodeName,omitempty" protobuf:"bytes,1,opt,name=nodeName"`
 	// The generation realized by the Node.
 	Generation int64 `json:"generation,omitempty" protobuf:"varint,2,opt,name=generation"`
+	// The flag to mark the NetworkPolicy realization is failed on a Node.
+	RealizationFailure *bool `json:"realizationFailure,omitempty" protobuf:"bytes,3,opt,name=realizationFailure"`
```

Comment:
Why we cannot use `bool`?

Reply:
Originally my thought is the previous version doesn't contain this field, so if Controller is co-working with the agent in previous versions, the field is possible not provided, and a pointer can avoid unexpected protobuf decoding error.

Think deeper, we can change the type as "bool" and remove "omitempty" in the field json format. In this way, protobuf will always generate RealizationFailure field, and set with the default value "false", which is also compatible with the previous versions.